code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,344 @@
+	var/joules_per_port
+	var/avail_fuel = get_fuel(fuel_ports)
+	var/list/fueled_ports = list()
+	var/ports_used
+
+	if(joules_req > avail_fuel) //Not enough fuel in the system.
+		return FALSE
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in fuel_ports)
+		if(F.has_fuel())
+			fueled_ports += F
+
+	joules_per_port = (joules_req / fueled_ports.len)
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in fueled_ports)
+		if(F.use_fuel_joules(joules_per_port))
+			ports_used++
+
+	if(ports_used == fueled_ports.len)
+		return TRUE
+
+/obj/machinery/ftl_shunt/core/proc/get_charge_time()
+	return (ftl_computer.linked.vessel_mass * CHARGE_TIME_PER_TON) * time_multiplier
+
+/obj/machinery/ftl_shunt/core/Process()
+	if(stat & (BROKEN|NOPOWER) && charging)
+		cancel_shunt()
+
+	if(!fuel_ports.len)
+		find_ports()
+
+	if(charging)
+		if(use_power != POWER_USE_ACTIVE)
+			update_use_power(POWER_USE_ACTIVE)
+		if(world.time >= charge_time) //We've probably finished charging up.
+			charging = FALSE
+			if(use_fuel(required_fuel_joules))
+				execute_shunt()
+			else
+				cancel_shunt() //Not enough fuel for whatever reason. Cancel.
+		SSradiation.radiate(src, (active_power_usage / 1000))
+
+	if(!charging)
+		update_use_power(POWER_USE_IDLE)
+
+/obj/machinery/ftl_shunt/fuel_port
+	name = "FTL shunt fuel port"
+	desc = "A fuel port for an FTL shunt device."
+	icon_state = "empty"
+
+	var/list/fuels = list(/decl/material/gas/hydrogen/tritium = 25000, /decl/material/gas/hydrogen/deuterium = 25000, /decl/material/gas/hydrogen = 25000, /decl/material/solid/exotic_matter = 50000)
+	var/obj/item/fuel_assembly/fuel
+	var/obj/machinery/ftl_shunt/core/master
+	var/max_fuel = 0
+
+/obj/machinery/ftl_shunt/fuel_port/on_update_icon()
+	if(fuel)
+		icon_state = "full"
+
+/obj/machinery/ftl_shunt/fuel_port/Destroy()
+	if(master)
+		master.fuel_ports -= src
+	master = null
+	QDEL_NULL(fuel)
+
+/obj/machinery/ftl_shunt/fuel_port/attackby(var/obj/item/O, var/mob/user)
+	if(istype(O, /obj/item/fuel_assembly))
+		if(!fuel)
+			if(!do_after(user, 20, src))
+				return
+			if(user && !user.unEquip(O))
+				return
+			O.forceMove(src)
+			fuel = O
+			max_fuel = get_max_fuel()
+			update_icon()
+/obj/machinery/ftl_shunt/fuel_port/attack_hand(var/mob/user)
+	if(fuel)
+		to_chat(user, SPAN_NOTICE("You begin to remove the fuel assembly from [src]..."))
+		if(!do_after(user, 20, src))
+			return
+		fuel.forceMove(get_turf(user))
+		user.put_in_hands(fuel)
+		fuel = null
+		max_fuel = 0
+
+
+
+/obj/machinery/ftl_shunt/fuel_port/proc/has_fuel()
+	if(fuel)
+		return TRUE
+	else
+		return FALSE
+
+/obj/machinery/ftl_shunt/fuel_port/proc/get_max_fuel()
+	if(!fuel)
+		return 0
+
+	var/total_joules_amount
+
+	for(var/G in fuel.rod_quantities)
+		if(G in fuels)
+			var/joules_per_unit = fuels[G]
+			var/total_joules = 10000 * joules_per_unit
+			total_joules_amount += total_joules
+
+	return total_joules_amount
+
+/obj/machinery/ftl_shunt/fuel_port/proc/get_avaliable_fuel_joules()
```

Comment:
Seems very similar to above proc

Reply:
One gets maximum fuel, one gets current fuel.

Reply:
You could collapse both procs into a single proc with a bool.
````
/obj/machinery/ftl_shunt/fuel_port/proc/get_fuel_joules(var/get_fuel_maximum)
	if(fuel)
		for(var/G in fuel.rod_quantities)
			if(G in fuels)
				. += (get_fuel_maximum ? fuel.rod_quantities[G] : 10000) * fuels[G]
````